src/utils/source-search.js

```diff
@@ -28,10 +28,20 @@ function getSearchState(cm) {
  * @memberof utils/source-search
  * @static
  */
-function getSearchCursor(cm, query, pos) {
-  // If the query string is all lowercase, do a case insensitive search.
-  return cm.getSearchCursor(query, pos,
-    typeof query == "string" && query == query.toLowerCase());
+function getSearchCursor(
+  cm, query, pos, caseSensitive, wholeWord, regexMatch) {
+  let moddedQuery = query;
```

Comment:
So CodeMirror's `getSearchCursor` has this signature

`getSearchCursor(query, start, caseFold)`

* query - string or RegExp object
* start - starting position in document
* caseFold - only relevant with strings but toggles case sensitive search

So the reason I had a moddedQuery was to avoid changing the original, but looking over the code there isn't much reason for it with the modifications I do.